### DISCUSSION TITLE
Ethernet over USB

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#25] [Ideas]
I was thinking of network access and as lots of people are now wifi only, it seems a quick win to use usb to access wireless network through everybody's smartphone.

Maybe ethernet over usb is more accessible than writing a specific driver for all existing hardware and avoid the binary blob as it will be buried in the phone :-).

**Comment by VendelinSlezak** [comment]
Thank you, this is definitely great idea how to avoid wifi drivers and still have internet access on most of computers. I will take a look how would it be possible to write driver for that.

**Comment by defdefred** [comment]
Happy you like the ideas :-)
Linux implementation could help?
https://elixir.bootlin.com/linux/latest/source/drivers/net/usb/usbnet.c

It could also allow the use of usb 4G modem like huawei E8372h.

**Comment by defdefred** [comment]
It could also allow the use of personal mini router with openwrt: https://openwrt.org/docs/guide-user/network/wan/smartphone.usb.reverse.tethering to access ethernet or wifi lan...

I have this one https://openwrt.org/toh/nexx/wt3020 and it works great.
